src/odemis/driver/test/npmc_test.py

```diff
@@ -0,0 +1,203 @@
+}
+
+# arguments used for the creation of basic components
+if TEST_NOHW:
+    CONFIG["port"] = "/dev/fake"
+
+
+def add_coord(pos1, pos2):
+    """
+    Adds two coordinate dictionaries together and returns a new coordinate dictionary.
+    pos1: dict (axis name str) -> (float)
+    pos2: dict (axis name str) -> (float)
+    Returns ret
+        dict (axis name str) -> (float)
+    """
+    ret = {}
+    for an, v in pos1.items():
+        if an in pos2.keys():
+            ret[an] = v + pos2[an]
+
+    return ret
+
+
+def subtract_coord(pos1, pos2):
+    """
+    Subtracts two coordinate dictionaries together and returns a new coordinate dictionary.
+    pos1: dict (axis name str) -> (float)
+    pos2: dict (axis name str) -> (float)
+    Returns ret
+        dict (axis name str) -> (float)
+    """
+    ret = {}
+    for an, v in pos1.items():
+        if an in pos2.keys():
+            ret[an] = v - pos2[an]
+
+    return ret
+
+
+class TestNPMC(unittest.TestCase):
+    """
+    Tests cases for the Newport ESP 301 actuator driver
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        cls.dev = npmc.ESP(**CONFIG)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.dev.terminate()  # free up socket.
+
+    def assert_coord_almost_equal(self, pos1, pos2):
+        """
+        Asserts that two coordinate dictionaries are almost equal to 4 decimal places.
+        """
+        self.assertEqual(pos1.keys(), pos2.keys())
+        for an, v in pos1.items():
+            self.assertAlmostEqual(round(v, 4), round(pos2[an], 4))
+
+    def check_second_conversion(self, pos_s, pos_mm):
+        """
+        Checks if a position in "delay seconds" is equivalent to one in mm
+        """
+        self.assertEqual(pos_s.keys(), pos_mm.keys())
+        for an, v in pos_s.items():
+            self.assertAlmostEqual(v * npmc.C / 2.0, pos_mm[an], places=4)
+
+    def test_position_abs(self):
+        """
+        Test moving to an absolute position.
+        """
+        new_pos = {'x': 0.1e-10, 'y':0, 'z': 0}
+        f = self.dev.moveAbs(new_pos)
+        f.result()
+        self.assert_coord_almost_equal(self.dev.position.value, new_pos)
+
+        new_pos = {'x': 0.05314e-9, 'y': 0.0653e-9, 'z': 0.0128e-9}
+        f = self.dev.moveAbs(new_pos)
+        f.result()
+        self.assert_coord_almost_equal(self.dev.position.value, new_pos)
+
+    def test_position_rel(self):
+        """
+        Test moving to a relative position.
+        """
+        old_pos = self.dev.position.value
+        shift = {'x': 0, 'y':0, 'z': 0}
+        f = self.dev.moveRel(shift)
+        f.result()
+        self.assertAlmostEqual(old_pos, self.dev.position.value)
+
+        # Test another relative move
+        old_pos = self.dev.position.value
+        shift = {'x': 0.1e-9, 'y':-0.2e-9, 'z':-0.3e-9}
+        f = self.dev.moveRel(shift)
+        f.result()
+        self.assert_coord_almost_equal(add_coord(old_pos, shift), self.dev.position.value)
+
```

Comment:
You need to test cancelling the moves, queueing several moves in a row, and calling `stop()`. In general, see tmcm_test for inspiration.